app/lib/slack/parses_slash_args_string.rb

```diff
@@ -0,0 +1,25 @@
+require "shellwords"
+
+module Slack
```

Comment:
**suggestion**: I'm not sure this should be the target module/namespace for this parser. The intention of the `Slack` module is to encapsulate anything we write that directly interacts with the Slack gem. Perhaps this would fit better under a `app/lib/utils` folder?

Reply:
I'm good with another namespace. I kinda feel like the stuff that interacts with slack at all should be somehow named accordingly though. `ParsesSlashCommandArgs` by itself feels a little ambiguous and my brain wants to see it grouped under a "Slack" umbrella somehow.

Reply:
I think the idea there is that double up uses slack because we happen to. The actual command parsing doesn't care whether it's a discord bot, slack app, teams plugin, etc. Regardless of the interaction, we'd want the same parsing logic to run as I see it. If we used discord or integrated at all, we'd have a Discord namespace, for example, that encapsulates all that work.

Reply:
> I think the idea there is that double up uses slack because we happen to.
I think I had the same thought as you but reached a different conclusion. Because the payload is Slack specific, the parser is slack Specific. If we added Discord then there would be a unique discord parser. This distinction is largely because I don't necessarily believe that the UX would be the same between platforms. This parser, IMO, is inherently tied to what we expect to receive from Slack.

Reply:
Interesting 🤔. I'm not seeing why the payloads would need to be platform specific, or at least why we would want the parser to be platform specific. The way I'm visualizing it is that we'd have a platform specific command handler, but we'd want to transform what is given to us into something a general parser would use. I use transform very loosely because the parameters would, I think, always be the same string structure/format.

I could see it problematic if we used the parser as a `before_action` potentially, but I don't think we'd use it directly like that as a controller action. Happy to discuss more if we're still seeing different pages.